server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1924,6 +1935,23 @@ public Volume migrateVolume(MigrateVolumeCmd cmd) {
                                 throw new InvalidParameterValueException("Cannot migrate ROOT volume of a stopped VM to a storage pool in a different VMware datacenter");
                             }
                         }
+
```

Comment:
agree with @rafaelweingartner.  All methods should have a single responsibility -- it is critical to maintainability and testability.  While it is not possible to fix this problem all at once, we certainly shouldn't be adding to it.